src/main/webapp/dataset.xhtml

```diff
@@ -144,41 +144,65 @@
                                                             <li class="dropdown-header">#{bundle['dataset.accessBtn.header.download']} <span class="glyphicon glyphicon-download-alt"/></li>
                                                             <!-- NORMAL DOWNLOAD BUTTON (NO TABULAR FILES) -->
-                                                            <li jsf:rendered="#{!DatasetPage.versionHasTabular}">
+                                                            <li jsf:rendered="#{!DatasetPage.versionHasTabular and !(DatasetPage.sizeOfDatasetNumeric > settingsWrapper.zipDownloadLimit)}">                 
                                                                 <p:commandLink update="@form" actionListener="#{DatasetPage.validateAllFilesForDownloadArchival()}" styleClass="btn-download">
                                                                     #{bundle.download}
                                                                     <h:outputFormat value="#{bundle['dataset.accessBtn.download.size']}">
                                                                         <f:param value="#{DatasetPage.sizeOfDataset}" />
                                                                     </h:outputFormat>
                                                                 </p:commandLink>
                                                             </li>
+                                                            
+                                                            <li jsf:rendered="#{DatasetPage.canDownloadFiles()  and (DatasetPage.sizeOfDatasetOrigNumeric > settingsWrapper.zipDownloadLimit) }" style="margin-left: 3.5em;">
+                                                                <h:outputText value="#{bundle['dataset.accessBtn.too.big']}"  >                                           
+                                                                </h:outputText> 
+                                                            </li>
+
                                                             <!-- DOWNLOAD ORIGINAL BUTTON (TABULAR FILES PRESENT) -->
-                                                            <li jsf:rendered="#{DatasetPage.versionHasTabular and DatasetPage.sizeOfDatasetOrig != '0 B' }">
-                                                                <p:commandLink update="@form" actionListener="#{DatasetPage.validateAllFilesForDownloadOriginal()}" styleClass="btn-download">
+                                                            <li jsf:rendered="#{DatasetPage.versionHasTabular and DatasetPage.sizeOfDatasetOrig != '0 B'  and !(DatasetPage.sizeOfDatasetOrigNumeric > settingsWrapper.zipDownloadLimit) }">
+                                                                <p:commandLink update="@form" actionListener="#{DatasetPage.validateAllFilesForDownloadOriginal()}" 
```

Comment:
it probably make sense, but can you explain the reason behind this change (and the similar one below it?)

Reply:
We don't want to render the download all here if the size exceeds the zip limit. Instead we show a message pointing the user to use the files table to select files.

Reply:
Right, I guess I'm trying to understand why there's a rendered on the li but also the validate method called on the link.

Reply:
The validate method on the link does a bunch of things like check to see if you have permission to download the files and compare the size of files you are asking for to the zip limit (which is more for the case where you select from the files table, but I put it in one method to prevent a lot of duplicate code from the prior version.) Then depending on whether you have selected files for which you don't have permission or too large of a download shows different popups - which is why your suggestion that we put the logic for displaying popups in the page may be hard to accomplish. i'll see what I can do there.

Reply:
right, but then what is the rendering checking? I must be missing something, if it helps explain it, we could do a quick call later.

Re: the popups, we should be trying to minimize the popups as much as possible; i.e. single message popups that have just a close. I would suggest trying to see if you can use the validate message to set what the message is in the popup (making sure the page updates the popup) and then the code to show the popup happens on the page. Let me know if that makes sense.

Reply:
The rendering here is checking for whether there are tabular files because it shows multiple links if there are (original or archival downloads). Maybe there's a different approach to take here, but I didn't really change it much except to add the various size tests(and I did set booleans in the init so that the renders won't have to do the same calculation multiple times)
Re: popups I thought you were objecting to the method actually opening them. I can work with the idea of having a single popup where the message changes - we already do that because we list the files that you don't have permission to download.
And Tania asked for the "just close" popup for when the size is too big - it also includes a link to the Download API in the guides.